app/components/Form/AmountInput.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import CurrencyIcon from 'components/CurrencyIcon'
+import styles from './RequestForm.scss'
+
+class AmountInput extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.handleChange = this.handleChange.bind(this)
+    this.handleKeyDown = this.handleKeyDown.bind(this)
+  }
+
+  parseNumber(_value) {
+    let value = _value.replace(/[^0-9,.]/g, '')
+    let integer = null
+    let fractional = null
+
+    if (value == '') { value = '0' }
+    if (value*1.0 < 0) { value = '0.0' }
+
+    // simplify this, make flat if possible
+    // extract the parsing bit?
+    if (value.match(/^[0-9]*[.,][0-9]*$/)) {
+      [integer, fractional] = value.toString().split(/[.,]/)
+      if (!fractional || fractional == '') {
```

Comment:
hmm, is this conditional needed? if `fractional` evaluates to false or `fractional == ''` then the conditional below will evaluate to false anyways right?